spark-board-ui/src/useDagAnimation/useDagAnimation.js

```diff
@@ -71,7 +71,7 @@ export function useDagAnimation(nodes, edges, targetPositions, setNodes) {
 
         // wait for the next frame
         animationId.current = requestAnimationFrame(onFrame)
-    }
+    }, [state, nodeIdMapping, nodes, setNodes]);
```

Comment:
This function is calling `setNodes` which change the `nodes`, so it's a circular dependency

Reply:
You're right, reverting this part of the change